roadmap/implementers-guide/src/node/utility/network-bridge.md

```diff
@@ -6,55 +6,118 @@ The most notable challenge is coordinating and eliminating race conditions of pe
 
-The other component of the network bridge is which peer-set to use. Different peer-sets can be connected for different purposes. The network bridge is not generic over peer-set, but instead exposes two peer-sets that event producers can attach to: `Validation` and `Collation`. More information can be found on the documentation of the [`NetworkBridgeMessage`][NBM].
+
+The network bridge makes use of the peer-set feature, but is not generic over peer-set. Instead, it exposes two peer-sets that event producers can attach to: `Validation` and `Collation`. More information can be found on the documentation of the [`NetworkBridgeMessage`][NBM].
 
 ## Protocol
 
 Input: [`NetworkBridgeMessage`][NBM]
-Output: Varying, based on registered event producers.
+
+
+Output:
+	- [`AvailabilityDistributionMessage`][AvD]`::NetworkBridgeUpdateV1`
+	- [`BitfieldDistributionMessage`][BitD]`::NetworkBridgeUpdateV1`
+	- [`PoVDistributionMessage`][PoVD]`::NetworkBridgeUpdateV1`
+	- [`StatementDistributionMessage`][StmtD]`::NetworkBridgeUpdateV1`
+	- [`CollatorProtocolMessage`][CollP]`::NetworkBridgeUpdateV1`
 
 ## Functionality
 
-Track a set of all Event Producers, each associated with a 4-byte protocol ID and the `PeerSet` it is associated on.
+This network bridge sends messages of these types over the network.
+
+```rust
+enum ProtocolMessage<M> {
+	ProtocolMessage(M),
+	ViewUpdate(View),
+}
+```
+
+and instantiates this type twice, once using the [`ValidationProtocolV1`][VP1] message type, and once with the [`CollationProtocolV1`][CP1] message type.
+
+```rust
+type ValidationV1Message = ProtocolMessage<ValidationProtocolV1>;
+type CollationV1Message = ProtocolMessage<CollationProtocolV1>;
+```
+
+### Startup
+
+On startup, we register two protocols with the underlying network utility. One for validation and one for collation. We register only version 1 of each of these protocols.
+
+### Main Loop
+
+The bulk of the work done by this subsystem is in responding to network events, signals from the overseer, and messages from other subsystems.
+
+Each network event is associated with a particular peer-set.
 
-There are two types of network messages this sends and receives:
+### Overseer Signal: ActiveLeavesUpdate
 
-- ProtocolMessage(ProtocolId, Bytes)
-- ViewUpdate(View)
+The `activated` and `deactivated` lists determine the evolution of our local view over time. A `ProtocolMessage::ViewUpdate` is issued to each connected peer on each peer-set, and a `NetworkBridgeEvent::OurViewChange` is issued to each event handler for each protocol.
 
-Each of these network messages is associated with a particular peer-set. If we are connected to the same peer on both peer-sets, we will receive two `ViewUpdate`s from them every time they change their view.
+If we are connected to the same peer on both peer-sets, we will send the peer two view updates as a result.
```

Comment:
The base assumption here is, that both protocol versions operate completely independently? So this assumes one subsystem accepting both protocol versions will never exist, since that might cause two incompatible views being received for version `A` and `A+1`

Reply:
It's not about protocol versions here, but about whether a node is connected to the local node as a collator or as a fellow validator.
It indeed means that you can't implement any logic specific to the situation where you are connected to a node is both a collator and a validator, but I don't think that's a problem.

Reply:
Subsystems might be able to receive network bridge notifications based on both the collation and validation peer-sets. But we currently make no guarantee that the views provided by a peer on one peer-set are the same as the views provided by a peer on the other peer-set.

At the moment, we have no subsystem which bridges both peer-sets. We can cross that bridge when we come to it, maybe by adding guarantees about views.